### DISCUSSION TITLE
Only allow delete if row is not referenced

### DISCUSSION THREAD

**Discussion by Laecherlich** [pocketbase/pocketbase#4110] [Q&A] [answered]
Hey, how can I create a filter rule which only allows deletion of rows if another table does not reference this row.

**Comment by ganigeorgiev** [comment, marked as answer]
You can use the `@collection.*` rule and exclude the id with `!=` (the default behavior for the `@colleciton.*` is match-all). For example:

```js
// require no record in otherCollection to have yourRelField equal the record id you are deleting 
@collection.otherCollection.yourRelField != id
```

Alternatively and depending on your collections structure, if that another collection reference your record with a single `relation` field you can mark it as "Non-empty" and the API should throw an error without extra rules when attempting to delete the record.

**Reply by Laecherlich** [reply]
Works perfect, thank you. 
Is the default behaviour of `@colleciton.*` in the docs, maybe I missed it?

By the way, really nice job with PocketBase. I really enjoy working with it. Currently using it for my bachelor-thesis-project so I maybe need to ask a few for question in the future :)

**Reply by ganigeorgiev** [reply]
You can find a little more information about the `@collection.*` rule in the ["Filters syntax" section](https://pocketbase.io/docs/api-rules-and-filters/#filters-syntax).
We apply match-all conditions by default for all multi-valued rules, aka. rules that could resolve to more than 1 record (eg. often multiple `relation` field(s) references).
If you want "any/at-least-one-of" condition you'll have to prefix the operators with `?`. For example, if we used `@collection.otherCollection.yourRelField ?!= id` this will be incorrect in your case because the `otherCollection` could contain one or more records (but no guarantee for all) that satisfy the non-equal constraint (eg. reference to another id) and the delete operation will be allowed.
